common/thorhelper/thorsoapcall.cpp

```diff
@@ -50,6 +50,41 @@ using roxiemem::OwnedRoxieString;
 #define CONNECTION "Connection"
 
 unsigned soapTraceLevel = 1;
+static StringBuffer soapSepString;
+static CriticalSection soapCrit;
+
+void setSoapSepString(const char *_soapSepString)
+{
+    CriticalBlock b(soapCrit);
+    soapSepString.set(_soapSepString);
+}
+
+static void multiLineAppendReplace(StringBuffer &origStr, StringBuffer &newStr)
+{
+    if (origStr.isEmpty())
+        return;
+    const char *cursor = origStr;
+    while (cursor)
```

Comment:
should be while (*cursor) (or could be while (true) which it effectively is)

Reply:
fixed